server/isomorphic-routing/index.js

```diff
@@ -20,22 +20,23 @@ export function serverRouter( expressApp, setUpRoute, section ) {
 
 function combineMiddlewares( ...middlewares ) {
 	return function( req, res, next ) {
-		req.context = getEnhancedContext( req );
+		req.context = getEnhancedContext( req, res );
 		applyMiddlewares( req.context, ...middlewares, () => {
 			next();
 		} );
 	};
 }
 
 // TODO: Maybe merge into getDefaultContext().
-function getEnhancedContext( req ) {
+function getEnhancedContext( req, res ) {
 	return Object.assign( {}, req.context, {
 		isLoggedIn: req.cookies.wordpress_logged_in,
 		isServerSide: true,
 		path: req.url,
 		pathname: req.path,
 		params: req.params,
-		query: req.query
+		query: req.query,
+		res
```

Comment:
I think we should be able to do the filter validation on the server side with this addition, we'll just have to branch on `isServerSide` for the redirect call. I'll do a PR once this one has landed.